crates/ruff_python_ast/src/identifier.rs

```diff
@@ -0,0 +1,587 @@
+    }
+}
+
+impl Identifier for Alias {
+    /// Return the [`TextRange`] for the identifier defining an [`Alias`].
+    ///
+    /// For example, return the range of `x` in:
+    /// ```python
+    /// from foo import bar as x
+    /// ```
+    fn identifier(&self, locator: &Locator) -> TextRange {
+        if matches!(self.name.as_str(), "*") {
+            self.range()
+        } else if self.asname.is_none() {
+            // The first identifier is the module name.
+            IdentifierTokenizer::new(locator.contents(), self.range())
+                .next()
+                .expect("Failed to find alias identifier")
+        } else {
+            // The first identifier is the module name.
+            // The second identifier is the "as" keyword.
+            // The third identifier is the alias name.
+            IdentifierTokenizer::new(locator.contents(), self.range())
+                .nth(2)
+                .expect("Failed to find alias identifier")
+        }
+    }
+}
+
+impl Identifier for Pattern {
+    /// Return the [`TextRange`] of the identifier in the given pattern.
+    ///
+    /// For example, return the range of `z` in:
+    /// ```python
+    /// match x:
+    ///     # Pattern::MatchAs
+    ///     case z:
+    ///         ...
+    /// ```
+    ///
+    /// Or:
+    /// ```python
+    /// match x:
+    ///     # Pattern::MatchAs
+    ///     case y as z:
+    ///         ...
+    /// ```
+    ///
+    /// Or :
+    /// ```python
+    /// match x:
+    ///     # Pattern::MatchMapping
+    ///     case {"a": 1, **z}
+    ///         ...
+    /// ```
+    ///
+    /// Or :
+    /// ```python
+    /// match x:
+    ///     # Pattern::MatchStar
+    ///     case *z:
+    ///         ...
+    /// ```
+    fn identifier(&self, locator: &Locator) -> TextRange {
+        match self {
+            Pattern::MatchAs(ast::PatternMatchAs {
+                name: Some(_),
+                pattern,
+                range,
+            }) => {
+                if let Some(pattern) = pattern {
+                    // Identify `z` in:
+                    // ```python
+                    // match x:
+                    //     case Foo(bar) as z:
+                    //         ...
+                    // ```
+                    IdentifierTokenizer::starts_at(pattern.end(), locator.contents())
+                        .nth(1)
+                        .expect("Unable to identify identifier in pattern")
+                } else {
+                    // Identify `z` in:
+                    // ```python
+                    // match x:
+                    //     case z:
+                    //         ...
+                    // ```
+                    *range
+                }
+            }
+            Pattern::MatchMapping(ast::PatternMatchMapping {
+                patterns,
+                rest: Some(_),
+                ..
+            }) => {
+                if let Some(pattern) = patterns.last() {
```

Comment:
```suggestion
                // From https://docs.python.org/3/reference/compound_stmts.html#grammar-token-python-grammar-mapping_pattern
                // > At most one double star pattern may be in a mapping pattern.
                // > The double star pattern must be the last subpattern in the mapping pattern.
                if let Some(pattern) = patterns.last() {
```

You might want to change the exact comment, but i expect many readers won't be familiar with what exactly the  `PatternMatchMapping` node entails